inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convert_color/convert_color_kernel_base.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "convert_color_kernel_base.h"
+#include "kernel_selector_utils.h"
+#include <string>
+
+namespace kernel_selector {
+
+bool ConvertColorKernelBase::Validate(const Params& p, const optional_params& o) const {
+    if (p.GetType() != KernelType::CONVERT_COLOR ||
+        o.GetType() != KernelType::CONVERT_COLOR) {
+        return false;
+    }
+
+    const convert_color_params& params = static_cast<const convert_color_params&>(p);
+
+    if (params.inputs[0].Dimentions() > 4)
+        return false;
+
+    return true;
+}
+
+CommonDispatchData ConvertColorKernelBase::SetDefault(const convert_color_params& params, const optional_params&) const {
+    const auto& out = params.output;
+    CommonDispatchData dispatchData;
+
+    dispatchData.gws = { out.Batch().v, out.Feature().v, out.Y().v * out.X().v };
+    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
```

Comment:
It's better from perf point to add info about in and out layout into `GetOptimalLocalWorkGroupSizes`, please find examples in the code.